storage/factory.go

```diff
@@ -45,6 +47,10 @@ type Factory interface {
 
 	// CreateDependencyReader creates a dependencystore.Reader.
 	CreateDependencyReader() (dependencystore.Reader, error)
+
+	// CreateLockAndSamplingStore creates a distributedlock.Lock and samplingstore.Store.
+	//  These are used for adaptive sampling.
```

Comment:
Sampling store could be static though, can it not? So not always for adaptive.

Reply:
The `SamplingStore` is explicitly for adaptive sampling. It is for storing and retrieving the various datapoints necessary to calculate the current sampling rates:

https://github.com/jaegertracing/jaeger/blob/master/storage/samplingstore/interface.go#L25